### PR TITLE
Use the BoM for all dependencies

### PR DESCRIPTION
**Pull request by samtstern**
We're making this the recommended pattern

### CODE REVIEW

File: functions/app/build.gradle [hunk 1]
```diff
@@ -36,19 +36,23 @@ dependencies {
...
     implementation 'com.google.android.material:material:1.2.1'
 
+    // Import the Firebase BoM (see: https://firebase.google.com/docs/android/setup)
+    implementation platform('com.google.firebase:firebase-bom:25.10.0')
+
     // Functions (Java)
-    implementation 'com.google.firebase:firebase-functions:19.1.0'
+    implementation 'com.google.firebase:firebase-functions'
 
     // Functions (Kotlin)
```
Comment (rachelsaunders) [inline functions/app/build.gradle:45]:
```suggestion
    // Cloud Functions for Firebase (Kotlin)
```

File: functions/app/build.gradle [hunk 2]
```diff
@@ -36,19 +36,23 @@ dependencies {
...
     // Firebase Authentication (Kotlin)
-    implementation 'com.google.firebase:firebase-auth-ktx:19.4.0'
+    implementation 'com.google.firebase:firebase-auth-ktx'
 
-    implementation 'com.google.firebase:firebase-messaging:20.2.4'
+    // Firebase Cloud Messaging
+    implementation 'com.google.firebase:firebase-messaging'
 
     implementation 'com.firebaseui:firebase-ui-auth:6.3.0'
     implementation 'com.google.android.gms:play-services-auth:18.1.0'
```
Comment (rachelsaunders) [inline functions/app/build.gradle:58]:
```suggestion
    // Firebase UI
    implementation 'com.firebaseui:firebase-ui-auth:6.3.0'
    
    // Google Play services
    implementation 'com.google.android.gms:play-services-auth:18.1.0'
```

File: mlkit/app/build.gradle [hunk 3]
```diff
@@ -45,8 +45,11 @@ dependencies {
     implementation 'androidx.media:media:1.2.0'
     implementation 'com.google.android.material:material:1.2.1'
 
+    // Import the Firebase BoM (see: https://firebase.google.com/docs/android/setup)
+    implementation platform('com.google.firebase:firebase-bom:25.10.0')
+
     // ML Kit dependencies
```
Comment (rachelsaunders) [inline mlkit/app/build.gradle:51]:
```suggestion
    // FirebaseML
```

File: perf/app/build.gradle [hunk 4]
```diff
@@ -37,7 +37,10 @@ dependencies {
     implementation project(":internal:lintchecks")
     implementation project(":internal:chooserx")
 
-    implementation 'com.google.firebase:firebase-perf:19.0.8'
+    // Import the Firebase BoM (see: https://firebase.google.com/docs/android/setup)
+    implementation platform('com.google.firebase:firebase-bom:25.10.0')
+
+    implementation 'com.google.firebase:firebase-perf'
```
Comment (rachelsaunders) [inline perf/app/build.gradle:43]:
```suggestion
    // Firebase Performance Monitoring
    implementation 'com.google.firebase:firebase-perf'
```

File: storage/app/build.gradle [hunk 5]
```diff
@@ -31,17 +31,20 @@ dependencies {
     implementation project(":internal:lintchecks")
     implementation project(":internal:chooserx")
 
+    // Import the Firebase BoM (see: https://firebase.google.com/docs/android/setup)
+    implementation platform('com.google.firebase:firebase-bom:25.10.0')
+
     // Storage (Java)
```
Comment (rachelsaunders) [inline storage/app/build.gradle:37]:
```suggestion
    // Cloud Storage for Firebase (Java)
```

File: storage/app/build.gradle [hunk 6]
```diff
@@ -31,17 +31,20 @@ dependencies {
...
-    implementation 'com.google.firebase:firebase-storage:19.2.0'
+    implementation 'com.google.firebase:firebase-storage'
 
     // Storage (Kotlin)
```
Comment (rachelsaunders) [inline storage/app/build.gradle:40]:
```suggestion
    // Cloud Storage for Firebase (Kotlin)
```

**Review by rachelsaunders — CHANGES_REQUESTED** [review]
nits about consistency for product names

File: admob/app/build.gradle [hunk 7]
```diff
@@ -43,9 +43,12 @@ dependencies {
     implementation 'com.google.android.gms:play-services-ads:19.4.0'
     // [END gradle_play_config]
 
+    // Import the Firebase BoM (see: https://firebase.google.com/docs/android/setup)
+    implementation platform('com.google.firebase:firebase-bom:25.10.0')
```
[the remark this answers is not in the document]
Comment (samtstern) [reply admob/app/build.gradle:47]: We've got a bot that does it, it should get auto-updated daily if needed.

File: admob/app/build.gradle [hunk 8]
```diff
@@ -43,9 +43,12 @@ dependencies {
     implementation 'com.google.android.gms:play-services-ads:19.4.0'
     // [END gradle_play_config]
 
+    // Import the Firebase BoM (see: https://firebase.google.com/docs/android/setup)
```
[the remark this answers is not in the document]
Comment (samtstern) [reply admob/app/build.gradle:46]: Yes that's better!

**Review by rachelsaunders — APPROVED** [review]
Pls do not submit until the new "/learn-more" page is live (~end of Sept)

**Pull request merged**